app/scripts/contentscript.js

```diff
@@ -82,6 +80,51 @@ function injectScript(content) {
   }
 }
 
+/**
+ * SERVICE WORKER LOGIC
+ */
+
+const WORKER_KEEP_ALIVE_INTERVAL = 1000;
+const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
+const TIME_45_MIN_IN_MS = 45 * 60 * 1000;
+
```

Comment:
We should keep service worker alive even if account is not connected on DAPP I think.  There are some queries that user can make even in disconnected state.

Reply:
Slightly similar point to https://github.com/MetaMask/metamask-extension/pull/16075#discussion_r1011118652, something to keep in mind when settling the above:

If the user is interacting with MetaMask on a dapp on one page, this should not result in events/change of behavior readable from an unauthorized (disconnected) tab/page as this would both leak entropy facilitating fingerprinting as well as help in detecting user activity (e.g distinguishing from "visiting browser has MetaMask installed" vs "visiting user has opened MetaMask during this session".

Reply:
@jpuri 
> There are some queries that user can make even in a disconnected state.
I'm not sure I know what queries you mean here. Is this something you can elaborate on? 

The keep-alive logic starts for any dapp interaction, whether or not accounts are connected. On initial load, we only trigger the keep-alive logic if the dapp connects or if the dapp sends requests. Initially, I don't think there is a way to distinguish a dapp from a non-dapp page aside from connected accounts... unless I'm missing something. I actually prefer to avoid running the keep-alive logic on non-dapps or when the provider is not in use on a page to save CPU and memory on the browser.

@legobeat getting to these comments after I run some more tests sending messages to/from our content script and extension

Reply:
@legobeat I added a reply here https://github.com/MetaMask/metamask-extension/pull/16075#discussion_r1013903709